### PR TITLE
Use ATS 8.1 in CDN in a Box

### LINKED ISSUE #5918: ATS in CiaB: /usr/bin/traffic_cop: error while loading shared libraries: libhwloc.so.5
## I'm submitting a ...

-  bug report

## Traffic Control components affected ...

-  CDN in a Box

## Current behavior:

Accompanying the June 3 CentOS 8.4.2105 release, `hwloc-libs` and other libraries updated, breaking compatibility with the 2018 `trafficserver-7.1.4-2.el7.x86_64` RPM that we use for the caches in the CDN in a Box. Attempting to run `trafficserver` manually results in a shared library error being logged to `/var/log/trafficserver/traffic_server.stderr`:

```
/usr/bin/traffic_cop: error while loading shared libraries: libhwloc.so.5: cannot open shared object file: No such file or directory
```

This breaks the *CDN-in-a-Box CI* GitHub Actions workflow.

## Expected behavior:

Traffic Server should run without any shared library issues.

## Minimal reproduction of the problem with instructions:

Run `traffic_cop` to see the shared library error:
```shell
docker-compose build --no-cache --parallel edge
docker-compose run --rm edge traffic_cop
```

## Anything else:

*   Installing `hwloc-libs-1.11.9-3.el8.x86_64` (previous version) instead of `hwloc-libs-2.2.0-1.el8.x86_64` (current version) is not an option, the package is not found:
    ```
    [user@computer trafficcontrol]$ docker run --rm centos:8.3.2011 dnf install hwloc-libs-1.11.9-3.el8.x86_64
    CentOS Linux 8 - AppStream                      7.1 MB/s | 7.4 MB     00:01
    CentOS Linux 8 - BaseOS                         3.0 MB/s | 2.6 MB     00:00
    CentOS Linux 8 - Extras                          30 kB/s | 9.6 kB     00:00
    No match for argument: hwloc-libs-1.11.9-3.el8.x86_64
    Error: Unable to find a match: hwloc-libs-1.11.9-3.el8.x86_64
    ```

*   The Dockerfile already [symlinks disparate library versions](https://github.com/apache/trafficcontrol/blob/5ad059f0af/infrastructure/cdn-in-a-box/cache/Dockerfile#L59-L68) for `libtcl`, `libncursesw`, and `libtinfo`. If `libhwloc.so.15` is symlinked to `libhwloc.so.5` and `traffic_cop` is started,

    ```shell
    ln -s libhwloc.so.15 /usr/lib64/libhwloc.so.5
    traffic_cop --stdout
    ```

    then `traffic_server` crashes (let's not pursue the CentOS 7 ATS 7 RPM on CentOS 8 any further).

### LINKED ISSUE #5919: CDN in a Box should use ATS 8.1
## I'm submitting a ...

-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...

-  CDN in a Box

## Current behavior:

CDN in a Box caches run `trafficserver-7.1.4-2.el7.x86_64`, even on CentOS 8.

## New behavior:

CDN in a Box should run Traffic Server 8.1.

## Minimal reproduction of the problem with instructions:

```shell
[user@computer cdn-in-a-box]$ docker-compose run --rm edge rpm -q trafficserver
Creating cdn-in-a-box_edge_run ... done
trafficserver-7.1.4-2.el7.x86_64
```

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5918 by using a Traffic Server RPM built for the latest version of CentOS 8 and getting the *CDN-in-a-Box CI* GitHub Actions workflow to pass.
- [x] This PR fixes #5919 by making CDN in a Box use a `trafficserver` RPM built from the 8.1.x ATS release branch.

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Documentation
- GitHub Actions
- T3C integration tests

## What is the best way to verify this PR?

1.  Build and run CDN in a Box, ensure caches are reported as healthy on http://trafficmonitor.infra.ciab.test/
    ```shell
    make -j6
    docker-compose build --parallel
    docker-compose kill && docker-compose down -v && docker-compose up -d
    docker-compose logs -f trafficmonitor
    ```

2.  Build and run CDN in a Box for CentOS 7, ensure caches are reported as healthy on http://trafficmonitor.infra.ciab.test/
    ```shell
    export RHEL_VERSION=7
    make -j6
    docker-compose build --parallel
    docker-compose kill && docker-compose down -v && docker-compose up -d
    docker-compose logs -f trafficmonitor
    ```

3.  Check the result of the *CDN-in-a-Box CI* GitHub Actions workflow

## If this is a bug fix, what versions of Traffic Control are affected?

 
 - master (5ad059f0af)
 - 5.1.2

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: infrastructure/cdn-in-a-box/Makefile [hunk 1]
```diff
@@ -164,6 +180,9 @@ $(ORT_RPM_ABSOLUTE): $(ORT_DIST_RPM)
 	cp -f "$?" "$@"
 
 # Dist rpms
+$(ATS_DIST_RPM): $(ATS_SOURCE)
+	docker-compose -f ../../cache-config/testing/docker/docker-compose-ats-build.yml build --parallel trafficserver_build && docker-compose -f ../../cache-config/testing/docker/docker-compose-ats-build.yml run --rm trafficserver_build
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/Makefile:184]: Don't these `../../`s need to be expanded from some variable?

File: infrastructure/cdn-in-a-box/bin/ats-version.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+set -o errexit -o nounset -o pipefail
+
+project=trafficserver
+script_dir="$(dirname "$0")"
+ats_version_file="${script_dir}/../cache/ATS_VERSION"
+
+remote_ats_version() {
+  local gitbox_url=https://gitbox.apache.org/repos/asf
+  local repo="${project}.git"
+	local branch refs commit last_tag release
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/bin/ats-version.sh:29]: Inconsistent indentation. You can use whatever you want as long as it's consistent, but just remember that tabs are inherently superior and if you use spaces you're a bad person.

File: infrastructure/cdn-in-a-box/traffic_ops_data/profiles/010-ATS_EDGE_TIER_CACHE.json [hunk 3]
```diff
@@ -87,7 +87,7 @@
 			"configFile": "records.config",
 			"name": "CONFIG proxy.config.log.logfile_dir",
 			"secure": false,
-			"value": "STRING /var/log/trafficserver"
+			"value": "STRING /opt/trafficserver/var/log/trafficserver"
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/traffic_ops_data/profiles/010-ATS_EDGE_TIER_CACHE.json:90]: Shouldn't this be determined from the RPM the same way the configuration files are for the `chown` step?

Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_data/profiles/010-ATS_EDGE_TIER_CACHE.json]: Getting logfile_dir from the RPM in 4486105741

File: infrastructure/cdn-in-a-box/bin/ats-version.sh (hunk 2 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/bin/ats-version.sh]: Using good-person spacing in a2a18f830b

File: infrastructure/cdn-in-a-box/Makefile (hunk 1 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/Makefile]: Expanding from some variable in fdb6f38212.

File: .github/workflows/ciab.yaml [hunk 4]
```diff
@@ -215,16 +215,52 @@ jobs:
...
+          path: ${{ github.workspace }}/dist
+          key: ${{ steps.git-repo-sha.outputs.sha }}-el${{ env.RHEL_VERSION }}-${{ hashFiles('cache-config/testing/docker/trafficserver/**') }}
+      - name: Build ATS RPM
+        if: steps.ats-rpm-cache.outputs.cache-hit != 'true'
+        uses: ./.github/actions/build-ats-test-rpm
+        env:
+          ATC_COMPONENT: ${{ github.job }}
+      - name: Display structure of dist directory
+        run: ls -lR
+        working-directory: ${{ github.workspace }}/dist
```
[the remark this answers is not in the document]
Comment (zrhoffman) [reply .github/workflows/ciab.yaml:244]: I just copy/pasted the `trafficserver` job from the [*Traffic Control Cache Config integration tests*](https://github.com/apache/trafficcontrol/blob/6f71cfa2dcd6f4684d423c22187d5e2662916a25/.github/workflows/cache-config-tests.yml#L93-L126) workflow. Removed the *Display SHA* step from both workflows in b4d175c802.

File: infrastructure/cdn-in-a-box/cache/run.sh [hunk 5]
```diff
@@ -89,6 +126,6 @@ chmod "0644" "/etc/cron.d/traffic_ops_ort-cron" && crontab "/etc/cron.d/traffic_
 
 crond -im off
 
-touch /var/log/trafficserver/diags.log
+touch /opt/trafficserver/var/log/trafficserver/diags.log
 # Leaves the container hanging open in the event of a failure for debugging purposes
-tail -Fn +1 /var/log/trafficserver/diags.log /var/log/ort.log
+tail -Fn+1 /opt/trafficserver/var/log/trafficserver/diags.log /var/log/ort.log
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/cache/run.sh:131]: Should these also be determined from the rpm? Not the ORT log file, that's probably fine, just the ATS  stuff

Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/cache/run.sh]: Using the gotten logfile_dir value here has well in b1ef527c1e.

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**